src/Uno.UI/UI/Xaml/Media/ImageSource.iOS.cs

```diff
@@ -190,44 +202,65 @@ internal async Task<UIImage> Open(CancellationToken ct)
 		/// </summary>
 		private async Task OpenUsingPlatformDownloader(CancellationToken ct)
 		{
-			if (SupportsAsyncFromBundle)
-			{
-				DownloadUsingPlatformDownloader();
-			}
-			else
+			if (ct.IsCancellationRequested)
 			{
-				await CoreDispatcher.Main.RunAsync(
-					CoreDispatcherPriority.Normal,
-					() =>
-					{
-						DownloadUsingPlatformDownloader();
-					}
-				).AsTask(ct);
+				return;
 			}
-		}
 
-		private void DownloadUsingPlatformDownloader()
-		{
 			using (var url = new NSUrl(WebUri.OriginalString))
 			{
-				NSError error;
-
-				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				using (var request = NSUrlRequest.FromUrl(url))
 				{
-					this.Log().Debug($"Loading image from [{WebUri.OriginalString}]");
-				}
+					NSUrlSessionDataTask task;
+					var awaitable = DefaultSession.CreateDataTaskAsync(request, out task);
+					ct.Register(OnCancel);
+					try
+					{
+						task.Resume(); // We need to call this manually https://bugzilla.xamarin.com/show_bug.cgi?id=28425#c3
+						var result = await awaitable;
+						task = null;
```

Comment:
What's this for ?

Reply:
Partly to avoid a leak if the CT is long-lived, partly so that cancelling the CT after the download completes is a no-op. (Mainly caution in other words.)